be/src/exec/pipeline/exchange/exchange_source_operator.cpp

```diff
@@ -32,14 +26,31 @@ bool ExchangeSourceOperator::is_finished() const {
 
 void ExchangeSourceOperator::set_finishing(RuntimeState* state) {
     _is_finishing = true;
-    return _stream_recvr->close();
+    static_cast<ExchangeSourceOperatorFactory*>(_factory)->close_stream_recvr();
 }
 
 StatusOr<vectorized::ChunkPtr> ExchangeSourceOperator::pull_chunk(RuntimeState* state) {
     std::unique_ptr<vectorized::Chunk> chunk = std::make_unique<vectorized::Chunk>();
-    RETURN_IF_ERROR(_stream_recvr->get_chunk(&chunk));
+    RETURN_IF_ERROR(_stream_recvr->get_chunk_pipeline(&chunk));
     eval_runtime_bloom_filters(chunk.get());
     return std::move(chunk);
 }
 
+std::shared_ptr<DataStreamRecvr> ExchangeSourceOperatorFactory::create_stream_recvr(
+        RuntimeState* state, const std::shared_ptr<RuntimeProfile>& profile) {
+    if (_stream_recvr != nullptr) {
+        return _stream_recvr;
+    }
+    _stream_recvr = state->exec_env()->stream_mgr()->create_recvr(
```

Comment:
One Fragment instance can only own one `DataStreamRecvr`, because there is a `_receiver_map` mapping `<fragment-id>-<dest-plan-node-id>` to `DataStreamRecvr`.

Multiple `ExchangeSource` should use one shared `DataStreamRecvr`.

```C++
std::shared_ptr<DataStreamRecvr> DataStreamMgr::create_recvr(
        RuntimeState* state, const RowDescriptor& row_desc, const TUniqueId& fragment_instance_id,
        PlanNodeId dest_node_id, int num_senders, int buffer_size, const std::shared_ptr<RuntimeProfile>& profile,
        bool is_merging, std::shared_ptr<QueryStatisticsRecvr> sub_plan_query_statistics_recvr, bool is_pipeline) {
    DCHECK(profile != nullptr);
    VLOG_FILE << "creating receiver for fragment=" << fragment_instance_id << ", node=" << dest_node_id;
    std::shared_ptr<DataStreamRecvr> recvr(
            new DataStreamRecvr(this, state, row_desc, fragment_instance_id, dest_node_id, num_senders, is_merging,
                                buffer_size, profile, std::move(sub_plan_query_statistics_recvr), is_pipeline));
    uint32_t hash_value = get_hash_value(fragment_instance_id, dest_node_id);
    std::lock_guard<std::mutex> l(_lock);
    _fragment_stream_set.emplace(fragment_instance_id, dest_node_id);
    _receiver_map.insert(std::make_pair(hash_value, recvr));
    return recvr;
}
```